cirq-core/cirq/transformers/noise_adding.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Mapping
+
+from cirq import ops, circuits
+from cirq.transformers import transformer_api
+import numpy as np
+
+
+def _gate_in_moment(gate: ops.Gate, moment: circuits.Moment) -> bool:
+    """Check whether `gate` is in `moment`."""
+    target_gate_in_moment = False
+    for op in moment.operations:
+        if op.gate == gate:
+            target_gate_in_moment = True
+            break
+    return target_gate_in_moment
+
+
+@transformer_api.transformer
+class DepolerizingNoiseTransformer:
+    """Add local depolarizing noise after two-qubit gates in a specified circuit. More specifically,
+    with probability p, append a random non-identity two-qubit Pauli operator after each specified
+    two-qubit gate.
+
+    Attrs:
+        p: The probability with which to add noise.
+        target_gate: Add depolarizing nose after this type of gate
+        rng: The pseudorandom number generator to use.
+    """
+
+    def __init__(
+        self,
+        p: float | Mapping[tuple[ops.Qid, ops.Qid], float],
+        target_gate: ops.Gate = ops.CZ,
+        rng: np.random.Generator | None = None,
+    ):
+        if rng is None:
+            rng = np.random.default_rng()
+        self.p = p
+        self.target_gate = target_gate
+        self.rng = rng
+
+    def __call__(
+        self,
+        circuit: circuits.AbstractCircuit,
+        *,
+        context: transformer_api.TransformerContext | None = None,
+    ):
+        """Apply the transformer to the given circuit.
+
+        Args:
+            circuit: The circuit to add noise to.
+            context: Not used; to satisfy transformer API.
+
+        Returns:
+            The transformed circuit.
+
+        Raises:
```

Comment:
can you move the validation of `p` to the constructor? ... this way you don't need to do it here